diamond/src/bin/diamond-server.rs

```diff
@@ -22,65 +24,57 @@ struct Args {
     generate: bool,
 }
 
-fn run(args: Args) -> Result<(), Error> {
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args = Args::from_args();
+
     if args.generate {
         Settings::generate(args.config.unwrap())?;
         exit(1);
     }
 
     let settings = Settings::new(args.config)?;
 
-    let tcp_addr = format!("{0}:{1}", &settings.tcp.host, settings.tcp.port).parse()?;
-    let udp_addr = format!("{0}:{1}", &settings.udp.host, settings.udp.port).parse()?;
+    let tcp_addr =
+        format!("{0}:{1}", &settings.tcp.host, settings.tcp.port).parse::<SocketAddr>()?;
```

Comment:
Consider type declaration instead of turbofish

Reply:
done